### PR TITLE
Fix false verbatimModuleSyntax error on `export namespace` in ESM

### LINKED ISSUE #52461: verbatimModuleSyntax not work with namespace at top level
# Bug Report

verbatimModuleSyntax not work with namespace

### 🔎 Search Terms

verbatimModuleSyntax

### 🕗 Version & Regression Information

5.0.0-beta

### ⏯ Playground Link

N/A because need a package.json

### 💻 Code

```json
{
    "type": "module",
        "dependencies": {
                "typescript": "5.0.0-beta"
        }
}
```

```ts
export namespace T {
        export enum A { a = 1 }
}
```

> npx tsc --moduleResolution NodeNext --module NodeNext --verbatimModuleSyntax .\index.mts

### 🙁 Actual behavior

```
index.mts:1:1 - error TS1287: A top-level 'export' modifier cannot be used on value declarations in a CommonJS module when 'verbatimModuleSyntax' is enabled.

1 export namespace T {
  ~~~~~~


Found 1 error in index.mts:1
```

### 🙂 Expected behavior

works

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #52461

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -43324,7 +43324,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         getNodeLinks(node).flags |= NodeCheckFlags.LexicalModuleMergesWithClass;
                     }
                 }
-                if (compilerOptions.verbatimModuleSyntax && node.parent.kind === SyntaxKind.SourceFile) {
+                if (compilerOptions.verbatimModuleSyntax &&
+                    node.parent.kind === SyntaxKind.SourceFile &&
+                    (moduleKind === ModuleKind.CommonJS || getSourceFileOfNode(node).impliedNodeFormat === ModuleKind.CommonJS)
```
Comment (jakebailey) [inline src/compiler/checker.ts:43329]: Ignore this if you want (I don't think it matters), but having just checked `node.parent.kind` you could probably just do `node.parent.impliedNodeFormat`.

File: tests/baselines/reference/verbatimModuleSyntaxRestrictionsESM(esmoduleinterop=false).errors.txt [hunk 2]
```diff
@@ -2,17 +2,6 @@
 /main.ts(3,8): error TS1259: Module '"/decl"' can only be default-imported using the 'allowSyntheticDefaultImports' flag
 
 
-==== /main.ts (2 errors) ====
```
Comment (jakebailey) [inline tests/baselines/reference/verbatimModuleSyntaxRestrictionsESM(esmoduleinterop=false).errors.txt:5]: So weird that the file's output moved in the baseline. It moved down to the bottom in this one, but up to the top in `.types`.

**Review by jakebailey — APPROVED** [review]

File: tests/baselines/reference/verbatimModuleSyntaxRestrictionsESM(esmoduleinterop=false).errors.txt (hunk 2 above)
Comment (andrewbranch) [reply tests/baselines/reference/verbatimModuleSyntaxRestrictionsESM(esmoduleinterop=false).errors.txt:5]: Yeah, I stared at that for a good 5 minutes trying to figure out if I had screwed something up.

File: src/compiler/checker.ts (hunk 1 above)
Comment (andrewbranch) [reply src/compiler/checker.ts:43329]: Good call, I just copied and pasted the condition from the other place this error is issued.

**Pull request merged**